service/history/task/cross_cluster_task_processor_test.go

```diff
@@ -0,0 +1,292 @@
+		TimerJitterCoefficient:     dynamicconfig.GetFloatPropertyFn(0.1),
+	}
+	s.fetcherOptions = &FetcherOptions{
+		Parallelism:            dynamicconfig.GetIntPropertyFn(3),
+		AggregationInterval:    dynamicconfig.GetDurationPropertyFn(time.Millisecond * 100),
+		TimerJitterCoefficient: dynamicconfig.GetFloatPropertyFn(0.5),
+	}
+	s.retryPolicy = backoff.NewExponentialRetryPolicy(time.Millisecond * 50)
+	s.retryPolicy.SetMaximumInterval(time.Millisecond * 100)
+	s.retryPolicy.SetMaximumAttempts(3)
+}
+
+func (s *crossClusterTaskProcessorSuite) TestCrossClusterTaskProcessorStartStop() {
+	taskFetchers := NewCrossClusterTaskFetchers(
+		constants.TestClusterMetadata,
+		s.mockShard.Resource.GetClientBean(),
+		s.fetcherOptions,
+		s.mockShard.Resource.GetLogger(),
+	)
+	taskProcessors := NewCrossClusterTaskProcessors(s.mockShard, s.mockProcessor, taskFetchers, s.processorOptions)
+	s.Len(taskProcessors, len(taskFetchers))
+
+	s.mockShard.Resource.RemoteAdminClient.EXPECT().GetCrossClusterTasks(gomock.Any(), gomock.Any()).Return(
+		&types.GetCrossClusterTasksResponse{
+			TasksByShard: map[int32][]*types.CrossClusterTaskRequest{int32(s.mockShard.GetShardID()): {}},
+		}, nil,
+	).AnyTimes()
+
+	taskFetchers.Start()
+	taskProcessors.Start()
+	taskProcessors.Stop()
+	taskFetchers.Stop()
+}
+
+func (s *crossClusterTaskProcessorSuite) TestRespondPendingTasks_Failed() {
+	s.testRespondPendingTasks(true)
+}
+
+func (s *crossClusterTaskProcessorSuite) TestRespondPendingTasks_Success() {
+	s.testRespondPendingTasks(false)
+}
+
+func (s *crossClusterTaskProcessorSuite) testRespondPendingTasks(failedToRespond bool) {
+	fetcher := newTaskFetcher(
+		cluster.TestAlternativeClusterName,
+		cluster.TestCurrentClusterName,
+		nil, nil, nil, nil,
+	)
+	processor := newCrossClusterTaskProcessor(s.mockShard, s.mockProcessor, fetcher, s.processorOptions)
+	numPendingTasks := 10
+	completedTasks := 6
+	futureSettables := make([]future.Settable, numPendingTasks)
+	for i := 0; i != numPendingTasks; i++ {
+		taskFuture, settable := future.NewFuture()
+		processor.pendingTasks[int64(i)] = taskFuture
+		futureSettables[i] = settable
+	}
+
+	for idx := range rand.Perm(numPendingTasks)[:completedTasks] {
+		futureSettables[idx].Set(types.CrossClusterTaskResponse{TaskID: int64(idx)}, nil)
+	}
+
+	s.mockShard.Resource.HistoryClient.EXPECT().RespondCrossClusterTasksCompleted(gomock.Any(), gomock.Any()).DoAndReturn(
+		func(_ context.Context, request *types.RespondCrossClusterTasksCompletedRequest) (*types.RespondCrossClusterTasksCompletedResponse, error) {
+			select {
+			case <-processor.shutdownCh:
+			default:
+				close(processor.shutdownCh)
+			}
+
+			s.Len(request.TaskResponses, completedTasks)
+			s.Equal(s.mockShard.Resource.GetClusterMetadata().GetCurrentClusterName(), request.TargetCluster)
+			s.Equal(s.mockShard.GetShardID(), int(request.GetShardID()))
+			// s.False(request.GetFetchNewTasks())
```

Comment:
Did you want to keep this?

Reply:
Yeah, will uncomment this after the idl change is landed.